detectable/src/main/java/com/blackduck/integration/detectable/detectables/opam/parse/OpamFileParser.java

```diff
@@ -0,0 +1,148 @@
+package com.blackduck.integration.detectable.detectables.opam.parse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileReader;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.Arrays;
+import java.util.HashSet;
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+public class OpamFileParser {
+
+    private static final String VERSION = "version";
+    private static final String DEPENDS = "depends";
+    private static final String NAME = "name";
+
+    public OpamFileParser() {
+    }
+
+    //have a separate parseData and parse method to use for parsing opam show command, as it gives output in file format
+    public OpamParsedResult parse(File opamFile) {
+        List<String> parsedOpamDependencies = new ArrayList<>();
+        String projectName = "";
+        String projectVersion = "";
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(opamFile))) {
+
+
+            List<String> fileData = reader.lines().collect(Collectors.toList());
+
+            Map<String, String> parsedOutput = parseData(fileData);
+
+            if(parsedOutput.containsKey(VERSION)) {
+                projectVersion = parsedOutput.get(VERSION); // get the version from parsed file
+            }
+
+            if(parsedOutput.containsKey(NAME)) {
+                projectName = parsedOutput.get(NAME); // get the project name from parsed File
+            }
+
+            if(parsedOutput.containsKey(DEPENDS)) {
+                parsedOpamDependencies = Arrays.asList(parsedOutput.get(DEPENDS).split(", ")); // get the depends section from parsed file
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException("There was an error while parsing the opam file.", e);
+        }
+
+        return new OpamParsedResult(projectName, projectVersion, parsedOpamDependencies, opamFile);
+    }
+
+    public Map<String, String> parseData(List<String> lines) {
+        Map<String, String> output = new HashMap<>();
+        Set<String> dependsSection = new HashSet<>();
+        Pattern pattern = Pattern.compile("\"([^\"]+)\"");
+        boolean inDependsSection = false;
+        boolean inVersionSection = false;
+
+
+        for (String line : lines) {
+            line = line.trim();
+            // if the line contains os or system level dependencies skip them
+            if (line.isEmpty() || line.startsWith("#") || line.contains("arch") || line.contains("win") ||
+                    line.contains("cc") || line.contains("os") || line.contains("system") || line.contains("ming")) {
```

Comment:
Correct me if I'm wrong but we should probably be more careful when it comes to discarding lines.
For example, consider the following package: https://opam.ocaml.org/packages/albatross/
Its name contains "os" and we would skip it here.

Reply:
Yes, you are correct, let me try to find an alternative for this.

Reply:
Ideally we want to skip lines which have os specific demands such as (if os != win32) or something like that so I have changed that condition somewhat to achieve that, I will wait for QA to test this and if it causes trouble, will remove that condition.

Reply:
Another thought: maybe it's better to default to including dependencies in the result instead of skipping them?
When there is an OS-specific demand, it doesn't necessarily mean that the dependency will be skipped.
It deepens on the customer, I suppose. I personally would prefer to see false positives than false negatives in my results as this decreases the chance of missing some license/vulnerability risks.

Reply:
I would test considering adding these dependencies, the reason why those were removed in the first place is circular dependencies. I have seen a lot of examples in Opam where this occurs and as a result, the recursive approach gets stuck in infinite loop.

This only happens in opam show approach while the tree does a good job of filtering dependencies out. In the docs, we are gonna mention to users that for most accurate results they should use opam version 2.2.0 and above.

Reply:
No, I think we should not split into 2 detectables, the other approach will not result in a LOW accuracy, its just the approach is little bit convoluted, but we have same approach for other detectors as well.

Reply:
I updated the approach to include all the system level dependencies in the BOM. This required minor tweaking to avoid parsing the versions that span across multiple lines.